model/src/ratio_as_decimal.rs

```diff
@@ -0,0 +1,139 @@
+use bigdecimal::BigDecimal;
+use num::bigint::Sign as Sign04;
+use num::BigRational;
+use num_bigint::{BigInt, Sign as Sign03};
+use serde::{de, Deserializer, Serializer};
+use serde_with::{DeserializeAs, SerializeAs};
+use std::fmt;
+use std::str::FromStr;
+
+pub struct DecimalBigRational;
+
+impl<'de> DeserializeAs<'de, BigRational> for DecimalBigRational {
+    fn deserialize_as<D>(deserializer: D) -> Result<BigRational, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserialize(deserializer)
+    }
+}
+
+impl<'de> SerializeAs<BigRational> for DecimalBigRational {
+    fn serialize_as<S>(source: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serialize(source, serializer)
+    }
+}
+
+pub fn serialize<S>(value: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let top_bytes = value.numer().to_bytes_le();
+    let top = BigInt::from_bytes_le(sign_04_to_03(top_bytes.0), &top_bytes.1);
+    let bottom_bytes = value.denom().to_bytes_le();
+    let bottom = BigInt::from_bytes_le(sign_04_to_03(bottom_bytes.0), &bottom_bytes.1);
+    let decimal = BigDecimal::from(top) / BigDecimal::from(bottom);
+    serializer.serialize_str(&decimal.to_string())
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<BigRational, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct Visitor {}
+    impl<'de> de::Visitor<'de> for Visitor {
+        type Value = BigRational;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            write!(
+                formatter,
+                "a BigRational encoded as a decimal encoded string"
+            )
+        }
+
+        fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            BigRational::from_str(&s).map_err(|err| {
+                de::Error::custom(format!(
+                    "failed to decode {:?} as decimal BigRational: {}",
+                    s, err
+                ))
+            })
+        }
+    }
+
+    deserializer.deserialize_str(Visitor {})
```

Comment:
I think the `Visitor` might be unnecessarily complicated, you could just:
```rust
BigRational::from_str(str::deserialize(deserializer)?).map_err(...)
```

Reply:
Unfortunately, `BigRational::from_str` only works for things like `a / b`... I think we are going to have to come up with something better here. Like find the index of the decimal. Multiply and divide by 10^{number of digits to the right}.